File: FJT.ReportDesigner/MySqlDBModel/report_parameter_setting_mapping.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace FJT.ReportDesigner.MySqlDBModel
{
    public class report_parameter_setting_mapping
    {
        [Key]
        public int id { get; set; }

        [StringLength(255)]
        public string reportParamName { get; set; }

        [StringLength(255)]
        public string dbColumnName { get; set; }

        [StringLength(255)]
        public string displayName { get; set; }

        [StringLength(255)]
        public string pageRouteState { get; set; }

        public bool isHiddenParameter { get; set; }

        [StringLength(255)]
        public string type { get; set; }

        public int? dataSourceId { get; set; }

        public string options { get; set; }

        public bool isDeleted { get; set; }

        public bool isDisplay { get; set; }

        public decimal displayOrder { get; set; }

        [StringLength(255)]
        public string createdBy { get; set; }

        [StringLength(255)]
        public string updatedBy { get; set; }

        [StringLength(255)]
        public string deletedBy { get; set; }

        public int? createByRoleId { get; set; }

        public int? updateByRoleId { get; set; }

        public int? deleteByRoleId { get; set; }

        public DateTime createdAt { get; set; }

        public DateTime? deletedAt { get; set; }

        public DateTime? updatedAt { get; set; }

        [ForeignKey("dataSourceId")]
        public virtual FixedEntityDataelement FixedEntityDataelement { get; set; }
    }
}

File: fjt.api/models/ReportParameterSettingMapping.js
const Sequelize = require('sequelize');

module.exports = (sequelize) => {
    const ReportParameterSettingMapping = sequelize.define('ReportParameterSettingMapping', {
        id: {
            allowNull: false,
            autoIncrement: true,
            primaryKey: true,
            type: Sequelize.INTEGER
        },
        reportParamName: {
            type: Sequelize.STRING(255),
            allowNull: false
        },
        dbColumnName: {
            type: Sequelize.STRING(255),
            allowNull: false
        },
        displayName: {
            type: Sequelize.STRING(255),
            allowNull: false
        },
        pageRouteState: {
            type: Sequelize.STRING(255),
            allowNull: false
        },
        dataSourceId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        isHiddenParameter: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        type: {
            type: Sequelize.STRING(255),
            allowNull: false
        },
        options: {
            type: Sequelize.TEXT,
            allowNull: true
        },
        isDisplay: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        displayOrder: {
            allowNull: true,
            type: Sequelize.DECIMAL(6, 2)
        },
        isDeleted: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        createdBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        updatedBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        deletedBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        createByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        updateByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        deleteByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        }
    }, {
        tableName: 'report_parameter_setting_mapping',
        paranoid: true
    });
    ReportParameterSettingMapping.associate = (models) => {
        ReportParameterSettingMapping.belongsTo(models.FixedEntityDataelement, {
            foreignKey: 'dataSourceId',
            as: 'FixedEntityDataelement'
        });
        ReportParameterSettingMapping.hasMany(models.ReportMasterParameter, {
            foreignKey: 'parmeterMappingid',
            as: 'reportMasterParameters'
        });
    };

    return ReportParameterSettingMapping;
};
